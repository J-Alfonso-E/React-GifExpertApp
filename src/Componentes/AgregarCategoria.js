import React, { useState } from "react";
import PropTypes from 'prop-types';

export const AgregarCategoria = ({EnviarCategoria}) => { //Se recibe la funcion desde GifExpertApp

    const [ValorEntrada, SetValor] = useState('');

    const CambioValor = (e) => {
        SetValor(e.target.value);
        //console.log('Evento Cambio de Valor');
    }

    const EnvioCategoria = (e) => {
        e.preventDefault();
        if(ValorEntrada.trim().length > 2){
            
            EnviarCategoria(cats => [ValorEntrada, ...cats,]); //Cuando se recibe una funcion de Hook el Estado puede ser accedido de esta manera y agregar la nueva categoria
            SetValor('');
        }

        
    }

    return (
        <form onSubmit={EnvioCategoria}>
            <h2>Agregar Categoria</h2>
            <p> {ValorEntrada} </p> {/* Valor de Prueba */}
                <input
                type='text'
                value = {ValorEntrada}
                onChange={CambioValor}
                >
                </input>
        </form>
    )
};

AgregarCategoria.propTypes = {
    EnviarCategoria : PropTypes.func.isRequired
}