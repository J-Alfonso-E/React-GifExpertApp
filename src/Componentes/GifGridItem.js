import React from "react";
import PropTypes from 'prop-types';


export const GifGridItem = ({title, url}) => {
    console.log({title, url});

    return (
        <div className='card animate__fadeInDown'>
            <img className='card animate__bounceIn' src={url} alt={title}></img>
            <p className='card'>{title}</p>
        </div>
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}