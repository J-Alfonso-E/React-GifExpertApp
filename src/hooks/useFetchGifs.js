import { useEffect, useState } from "react"
import { ObtGif } from "../Complementos/GifFetch";

export const useFetchGifs = ( categoria ) => {
    const [state, setState] = useState({
        data: [],
        loading: true
    });

    useEffect( () => {
        ObtGif(categoria)
        .then(imgs => {
            setTimeout ( () => {
                setState({
                    data: imgs,
                    loading: false
                })
            }, 3000);
        })
    }, [categoria])

    /*setTimeout( () => {
        setState({
            data: [1,2,3,4],
            loading: false
        })
    }, 3000);*/

    return state; //{data: [], loading: true}
}