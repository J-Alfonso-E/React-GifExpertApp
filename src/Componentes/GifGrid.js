import React from "react";
import { useFetchGifs } from "../hooks/useFetchGifs";
//import { ObtGif } from "../Complementos/GifFetch";
import { GifGridItem } from "./GifGridItem";

import PropTypes from 'prop-types';

export const GifGrid = ( {categoria}) => {

    const {data, loading} = useFetchGifs( categoria);

    return (
        <>
        <h3>{categoria}</h3>

        {loading && 'Cargando' } {/* Operador Ternario donde si es true hace lo primero en caso contrario no hace nada */}
        <div className='card-grid animate__bounceIn'>
            
            
                {data.map( img => (
                    <GifGridItem key={img.id} {...img} />
                ))
                }
        </div>
        </>
    )
}

GifGrid.propTypes = {
    categoria: PropTypes.string.isRequired
}