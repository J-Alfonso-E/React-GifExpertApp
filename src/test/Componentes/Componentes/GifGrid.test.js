import React from 'react';
import {shallow} from 'enzyme';
import { GifGrid } from '../../../Componentes/GifGrid';
import { useFetchGifs } from '../../../hooks/useFetchGifs';
jest.mock('../../../hooks/useFetchGifs');

describe('Pruebas de GifGrid', () => {
    //let wrapper = shallow(<GifGrid categoria={''} />);

    /*beforeEach(() =>{ //Cada Prueba que se realiza se ejecuta esta funcion
        wrapper = shallow(<GifGrid categoria={''} />);
    });*/
    test('Snapshot de GifGrid Vacio', () =>{
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        const wrapper = shallow(<GifGrid categoria={'Tales Arisen'} />);
        expect(wrapper).toMatchSnapshot();
    }); 

    test('Snapshot de GifGrid con un objeto', () =>{

        const img = [{
            id: '123',
            url: 'https://localhost/cosa.jpg',
            title: 'Titulo de Prueba'
        }]

        useFetchGifs.mockReturnValue({
            data: img,
            loading: false
        });
        const wrapper = shallow(<GifGrid categoria={'Tales Arisen'} />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(img.length);
        
    }); 

    

    
});