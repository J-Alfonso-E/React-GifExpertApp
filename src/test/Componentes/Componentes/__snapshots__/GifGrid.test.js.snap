// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de GifGrid Snapshot de GifGrid Vacio 1`] = `
<Fragment>
  <h3>
    Tales Arisen
  </h3>
  Cargando
   
  <div
    className="card-grid animate__bounceIn"
  />
</Fragment>
`;

exports[`Pruebas de GifGrid Snapshot de GifGrid con un objeto 1`] = `
<Fragment>
  <h3>
    Tales Arisen
  </h3>
   
  <div
    className="card-grid animate__bounceIn"
  >
    <GifGridItem
      id="123"
      key="123"
      title="Titulo de Prueba"
      url="https://localhost/cosa.jpg"
    />
  </div>
</Fragment>
`;
